tests/basic_relay.go

```diff
@@ -0,0 +1,302 @@
+
+	nonceB, err := subnetBInfo.ChainWSClient.NonceAt(ctx, fundedAddress, nil)
+	Expect(err).Should(BeNil())
+
+	log.Info("Packing teleporter message", "nonceA", nonceA, "nonceB", nonceB)
+	payload, err = teleporter.PackSendCrossChainMessageEvent(common.Hash(subnetBInfo.BlockchainID), teleporterMessage)
+	Expect(err).Should(BeNil())
+
+	data, err = teleporter.EVMTeleporterContractABI.Pack(
+		"sendCrossChainMessage",
+		teleporterTestUtils.SendCrossChainMessageInput{
+			DestinationChainID: subnetBInfo.BlockchainID,
+			DestinationAddress: teleporterMessage.DestinationAddress,
+			FeeInfo: teleporterTestUtils.FeeInfo{
+				ContractAddress: fundedAddress,
+				Amount:          big.NewInt(0),
+			},
+			RequiredGasLimit:        teleporterMessage.RequiredGasLimit,
+			AllowedRelayerAddresses: teleporterMessage.AllowedRelayerAddresses,
+			Message:                 teleporterMessage.Message,
+		},
+	)
+	Expect(err).Should(BeNil())
+
+	// Send a transaction to the Teleporter contract
+	tx := teleporterTestUtils.NewTestTeleporterTransaction(subnetAInfo.ChainIDInt, teleporterContractAddress, nonceA, data)
+
+	txSigner := types.LatestSignerForChainID(subnetAInfo.ChainIDInt)
+	signedTx, err := types.SignTx(tx, txSigner, fundedKey)
+	Expect(err).Should(BeNil())
+
+	// Sleep for some time to make sure relayer has started up and subscribed.
+	time.Sleep(15 * time.Second)
+	log.Info("Subscribing to new heads on destination chain")
+
+	newHeadsB := make(chan *types.Header, 10)
+	sub, err := subnetBInfo.ChainWSClient.SubscribeNewHead(ctx, newHeadsB)
+	Expect(err).Should(BeNil())
+	defer sub.Unsubscribe()
+
+	log.Info("Sending teleporter transaction", "destinationChainID", subnetBInfo.BlockchainID, "txHash", signedTx.Hash())
+	teleporterTestUtils.SendAndWaitForTransaction(ctx, subnetAInfo.ChainWSClient, signedTx)
+
+	receipt, err := subnetAInfo.ChainWSClient.TransactionReceipt(ctx, signedTx.Hash())
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	sendCrossChainMessageLog := receipt.Logs[0]
+	var sendEvent teleporterTestUtils.SendCrossChainMessageEvent
+	err = teleporter.EVMTeleporterContractABI.UnpackIntoInterface(&sendEvent, "SendCrossChainMessage", sendCrossChainMessageLog.Data)
+	Expect(err).Should(BeNil())
+	log.Info(fmt.Sprintf("Sent teleporter message: %#v", sendEvent))
+	teleporterMessageID := sendEvent.Message.MessageID
+
+	// Get the latest block from Subnet B
+	log.Info("Waiting for new block confirmation")
+	newHead := <-newHeadsB
+	log.Info("Received new head", "height", newHead.Number.Uint64())
+	blockHash := newHead.Hash()
+	block, err := subnetBInfo.ChainWSClient.BlockByHash(ctx, blockHash)
+	Expect(err).Should(BeNil())
+	log.Info(
+		"Got block",
+		"blockHash", blockHash,
+		"blockNumber", block.NumberU64(),
+		"transactions", block.Transactions(),
+		"numTransactions", len(block.Transactions()),
+		"block", block,
+	)
+	accessLists := block.Transactions()[0].AccessList()
```

Comment:
It's fine for now, but just want to point out that if we want to run this test in parallel with anything else, we'll need some better ways to pick out the correct warp message in the case two different tests include one in the same block.

Reply:
Yeah agreed. I don't think the overhead of such considerations is worth parallelizing these tests.